### PR TITLE
fix(cli): say review coverage gaps in the author's units, not chunk ids

### PR DESCRIPTION
**Pull request by wenshao**
## What this PR does

`/review`'s posted body rendered coverage disclosures with the run's own bookkeeping as subjects: bare chunk ids, unsorted, one subject per chunk. This PR makes the posted body say chunk-level gaps in the PR author's units instead, and stops the opener from certifying over a body that denies it.

Three changes, all at render time — the structural entries, the event caps, the caller-echo dedup and the stderr remediation still key on chunk ids, which is where the id is the selector a reader can act on:

- Coverage now returns the plan's chunk→files table. `DiffChunk.files` was already written into the plan JSON; the coverage type slice dropped it.
- `compose-review` renders chunk gaps through a new `describeChunkGap`: every planned chunk collapses to "the entire diff", a narrow gap with known files names the files, and anything wider is counted against the plan's total ("38 of the diff's 49 sections"). Applied to the receipt sentence, the uncoverable sentence (bare CLI entries only — caller-authored entries like `chunk 5 (src/big.min.js)` still render verbatim) and the grouped per-cause sentences.
- The COMMENT opener may no longer say "Reviewed." above a disclosure set that denies it: when no chunk is both covered and undisclosed — or no chunk universe could be read at all — it opens with "⚠️ This run could not certify that any of this diff was reviewed." A rewritten launch demonstrably read its chunk, so coverage alone is not the test; certified means covered with no disclosure against it.

## Why it's needed

The review posted on PR #7268 ([this comment](https://github.com/QwenLM/qwen-code/pull/7268#pullrequestreview-4757341042)) is the motivating incident. The run certified nothing — 11 chunks ran on rewritten prompts, the other 38 plus two roles were never launched with their built prompts — and the posted body enumerated all 49 chunk ids across two sentences, the first group in transcript order (`chunk 28, chunk 49, chunk 6, …`), while opening with "Reviewed. Suggestions are inline."

Two defects in one body: the opener certified the exact thing every following sentence took back, and the disclosure vocabulary was the operator's, not the author's — nothing on the PR page maps `chunk 28` to code, and the register comments in `coverage.ts` already require the posted gap text to be author-facing. The commands were scrubbed from that register; the nouns were not.

### How to verify

1. Re-run the #7268 shape (49 chunks, 11 rewritten, 38 unlaunched, Step 4/5 not launched verbatim) through `composeReview`. The body is now:

   > ⚠️ This run could not certify that any of this diff was reviewed. Suggestions are inline. Not reviewed: 11 of the diff's 49 sections — launched with a prompt that is not the one the CLI built. Not reviewed: 38 of the diff's 49 sections — its prompt was built, but no agent on record was launched with it. Not reviewed: Test coverage matrix (whole-diff) — … Not reviewed: reverse audit — … Not reviewed: verification — …

   against the previous ~250-word body that enumerated every id and opened "Reviewed."
2. Confirm a partial gap still names what the author owns: one unlaunched chunk over `src/a.ts` renders "the diff section covering src/a.ts", and the opener stays "Reviewed." while any chunk is certified.
3. Confirm the operator's channel is untouched: `remediation` still carries the exact `--chunk <id>` rebuild selectors, and `check-coverage`'s stderr output is unchanged.
4. `npx vitest run src/commands/review/` from `packages/cli` — 784 tests pass. `eslint --max-warnings 0` on the three changed files passes; `tsc -p packages/cli` reports no errors in `commands/review` (the pre-existing `acp-integration` errors are stale-dist noise, present on `main`).

### Evidence (Before & After)

Before (posted on #7268): "Reviewed. Suggestions are inline. Not reviewed: chunk 28, chunk 49, chunk 6, chunk 46, chunk 39, chunk 4, chunk 26, chunk 27, chunk 7, chunk 38, chunk 48 — launched with a prompt that is not the one the CLI built. Not reviewed: chunk 1, chunk 2, chunk 3, chunk 5, chunk 8, … chunk 47, Test coverage matrix (whole-diff), Agent 1c: Cross-file tracer — its prompt was built, but no agent on record was launched with it. …"

After: the body quoted in step 1 above — no chunk ids anywhere in the posted body, one phrase per cause, and an opener that matches the disclosures. No UI change, so screenshots are N/A.

### Tested on

|     OS     | Status |
| :--------: | :----: |
|  🍏 macOS  |   ⚠️ not tested   |
| 🪟 Windows |   ⚠️ not tested   |
|  🐧 Linux  |   ✅ tested   |

## Risk & Scope

- Main risk or tradeoff: an operator who reads only the posted PR body loses the chunk-id detail there; it remains on stderr (remediation and `check-coverage`), which is the channel the register comments assign it to. A plan written before chunks carried `files` falls back to counts — the field is optional in the coverage reader.
- Not validated / out of scope: the roster role labels (`Agent 1c: Cross-file tracer`) still render verbatim in the body — reworking those labels forks the register shared with stderr and is left for a follow-up, as is collapsing the near-duplicate reverse-audit/verification `rewritten` sentences.
- Breaking changes / migration notes: none. `CoverageFromTranscripts` gains an additive `plannedChunks` field; all body changes are phrasing.

<details>
<summary>中文说明</summary>

## 本 PR 做了什么

`/review` 发布到 PR 上的 review body 此前用运行时的内部记账作为披露主语：裸的 chunk 编号、乱序、每个 chunk 一个主语。本 PR 让公开 body 以 PR 作者的单位来陈述 chunk 级缺口，并禁止开场白与后文的披露自相矛盾。

三处改动都在渲染层——结构化条目、verdict 上限、caller 回显去重和 stderr 修复命令仍以 chunk id 为键，那里 id 才是读者可以执行的选择器：

- coverage 现在返回 plan 的 chunk→files 映射表。`DiffChunk.files` 本来就写进了 plan JSON；只是 coverage 的类型切片把它丢掉了。
- `compose-review` 通过新的 `describeChunkGap` 渲染 chunk 缺口：覆盖全部 planned chunk 时收敛为 "the entire diff"；文件已知且数量少的窄缺口直接点名文件；更宽的缺口按 plan 总数计数（"38 of the diff's 49 sections"）。应用于 receipt 句、uncoverable 句（仅 CLI 自己加入的裸条目——caller 提供的 `chunk 5 (src/big.min.js)` 这类条目仍原样渲染）以及按 cause 分组的句子。
- COMMENT 开场白不得再在否定它的披露之上写 "Reviewed."：当没有任何 chunk 同时满足"被读过且无披露指向它"——或根本读不到 chunk 全集时——开场改为 "⚠️ This run could not certify that any of this diff was reviewed."。rewritten launch 的 agent 确实读了它的 chunk，所以仅凭 coverage 不够；"认证"意味着已覆盖且无披露与之相悖。

## 为什么需要它

动机事件是发布在 PR #7268 上的[这条 review](https://github.com/QwenLM/qwen-code/pull/7268#pullrequestreview-4757341042)。那次运行什么都没认证——11 个 chunk 跑在改写过的 prompt 上，其余 38 个加两个 role 从未按构建的 prompt 启动——而发布的 body 用两句话枚举了全部 49 个 chunk 编号（第一组还是 transcript 顺序：`chunk 28, chunk 49, chunk 6, …`），开头却写着 "Reviewed. Suggestions are inline."。

一个 body 两个缺陷：开场白认证了后面每句话都在收回的东西；披露词汇是 operator 的而非作者的——PR 页面上没有任何东西能把 `chunk 28` 映射到代码，而 `coverage.ts` 的语域注释本来就要求公开 gap 文案面向作者。命令被挡住了，名词没有。

### 如何验证

1. 用 #7268 的形态（49 chunk、11 rewritten、38 未启动、Step 4/5 未按原样启动）重跑 `composeReview`，body 现在是上文英文部分第 1 步引用的样子——对比之前约 250 词、逐个枚举编号、以 "Reviewed." 开头的 body。
2. 确认部分缺口仍点名作者拥有的东西：一个未启动、覆盖 `src/a.ts` 的 chunk 渲染为 "the diff section covering src/a.ts"；只要还有 chunk 被认证，开场白保持 "Reviewed."。
3. 确认 operator 通道不受影响：`remediation` 仍携带精确的 `--chunk <id>` 重建选择器，`check-coverage` 的 stderr 输出不变。
4. 在 `packages/cli` 下 `npx vitest run src/commands/review/`——784 个测试全部通过；三个改动文件通过 `eslint --max-warnings 0`；`tsc -p packages/cli` 在 `commands/review` 下无错误（`acp-integration` 的既有报错是 stale-dist 噪音，`main` 上同样存在）。

### 证据（Before & After）

Before（发布在 #7268 上）：逐个枚举 49 个 chunk 编号的两句话，开头 "Reviewed. Suggestions are inline."。

After：英文部分第 1 步引用的 body——公开 body 中不再出现任何 chunk 编号，每个 cause 一个短语，开场白与披露一致。无 UI 变更，截图 N/A。

### 测试平台

|     OS     | 状态 |
| :--------: | :----: |
|  🍏 macOS  |   ⚠️ 未测试   |
| 🪟 Windows |   ⚠️ 未测试   |
|  🐧 Linux  |   ✅ 已测试   |

## 风险与范围

- 主要风险或权衡：只看 PR body 的 operator 会失去那里的 chunk-id 细节；细节保留在 stderr（remediation 与 `check-coverage`），这正是语域注释为它指定的通道。chunk 不携带 `files` 的旧 plan 回退到计数——coverage 读取端该字段是可选的。
- 未验证或范围外：roster 的 role label（`Agent 1c: Cross-file tracer`）仍原样渲染进 body——改写这些 label 会分叉与 stderr 共享的语域，留作后续；reverse-audit 与 verification 的 `rewritten` 近重复句子的合并同样留作后续。
- Breaking changes / migration notes：无。`CoverageFromTranscripts` 新增 `plannedChunks` 字段（增量式）；body 改动均为措辞层。

</details>

### CODE REVIEW

**Comment by gwinthis** [comment]
## Review & Local Verification Report

### 代码审查

**设计评价：以用户为中心的信息设计修复。** 本 PR 解决了 review 覆盖缺口报告中使用内部 chunk ID（"chunk 1, chunk 2"）的问题——这些 ID 对 PR 作者毫无意义。

**核心变更：**
1. **用文件路径替代 chunk ID**：`describeChunkGap()` 将 chunk 映射到其覆盖的文件（"the diff section covering src/a.ts"）
2. **全量缺口聚合**：所有 chunk 未覆盖时说 "the entire diff"，而非列举 49 个 chunk ID
3. **零认证警告**：当没有任何 chunk 被认证时，开头从 "Reviewed." 改为 "⚠️ This run could not certify that any of this diff was reviewed."

**关键修复（#7268 回归）：** 之前 body 以 "Reviewed. Suggestions are inline." 开头，然后列出所有未覆盖的 chunk——第一句就"认证"了后面每一句都在否定的东西。

**测试覆盖：** 大量测试更新 + 新增测试，验证：
- chunk ID 不出现在 body 中（`not.toMatch(/chunk \d/)`）
- 全量缺口说 "the entire diff"
- 零认证时不以 "Reviewed." 开头
- 部分认证时保留 "Reviewed." 开头

### 结论

**LGTM。** 将内部簿记语言转换为用户可理解的语言，同时修复了自相矛盾的认证声明。

**Review by yiliang114 — APPROVED** [review]
LGTM. Clean, well-motivated change — chunk ids are internal bookkeeping and the author needs their own units (files, or the diff itself). The `describeChunkGap` translation is correct across all edge cases: full-plan collapse, narrow fileNamed gaps, wide gaps, unknown-chunk poisoning, and no-plan fallback. The zero-certification opener closes the #7268 regression structurally.

Test coverage is thorough — 144 added lines covering every branch of `describeChunkGap` plus the opener's certified/partial/zero states.

_— qwen3.8-max-preview via Qwen Code /review_

**Review by doudouOUC — COMMENTED** [review]
Reviewed. Suggestions are inline. Not reviewed: coverage — no plan was given, so this run cannot show that any of the diff was read.

_— qwen3.7-max via Qwen Code /review_

File: packages/cli/src/commands/review/compose-review.ts [hunk 1]
```diff
@@ -590,16 +599,34 @@ export function composeReview(input: ComposeReviewInput): ComposeReviewResult {
...
+    // report) and render through the same translation as every other chunk
+    // gap; a caller's entry may already carry the file (`chunk 5
+    // (src/big.min.js)`) and renders verbatim — its structure is not ours to
+    // reparse.
+    const bareIds: number[] = [];
+    const callerNamed: string[] = [];
+    for (const e of uncoverable) {
+      const m = /^chunk (\d+)$/.exec(e);
+      if (m) bareIds.push(Number(m[1]));
+      else callerNamed.push(e);
```
Comment (doudouOUC) [inline packages/cli/src/commands/review/compose-review.ts:620]: **[Suggestion]** The bare-chunk-ID regex `/^chunk (\d+)$/` plus `Number(m[1])` appears three times in this file — here (uncoverable split), at the byReason split, and in the opener's disclosed-ID collection. The two split-render sites also duplicate the surrounding `bareIds`/`named` partition loop and the `shown`-array assembly.

**Failure scenario:** If the chunk subject format ever changes (e.g. `chunk-5`, `Chunk 5`, or additional parenthetical context), three sites need updating in lockstep — a missed update silently leaks raw chunk IDs in one sentence type while the others render file names.

Extract a `parseBareChunkId(s: string): number | null` helper, and for the two split-render sites a `splitChunkEntries(entries: string[]): { bareIds: number[]; named: string[] }` that uses it.

_— qwen3.7-max via Qwen Code /review_

File: packages/cli/src/commands/review/compose-review.ts [hunk 2]
```diff
@@ -590,16 +599,34 @@ export function composeReview(input: ComposeReviewInput): ComposeReviewResult {
...
   }
   if (uncoverable.length > 0) {
+    // The CLI's own entries are bare `chunk <id>` (pushed above, from the
+    // report) and render through the same translation as every other chunk
+    // gap; a caller's entry may already carry the file (`chunk 5
+    // (src/big.min.js)`) and renders verbatim — its structure is not ours to
+    // reparse.
+    const bareIds: number[] = [];
+    const callerNamed: string[] = [];
+    for (const e of uncoverable) {
```
Comment (doudouOUC) [inline packages/cli/src/commands/review/compose-review.ts:617]: **[Suggestion]** The uncoverable-chunk rendering path's bare-ID branch is never exercised by any integration test. All existing uncoverable tests pass caller-named entries like `'chunk 5 (src/big.min.js)'`; the bare format `chunk <id>` (produced by `uncoverable.push(prefix)` when `cov.uncoverableChunks` has entries no caller has named) is dead in the test suite.

**Failure scenario:** The underlying `describeChunkGap` is unit-tested, but there is no integration test verifying that a bare uncoverable entry renders as a file phrase rather than leaking a chunk ID through the full `composeReview` path.

Add one test that feeds an uncoverable entry without a caller-named variant alongside a plan whose chunks carry files, and assert the body contains the file name rather than `chunk 1`.

_— qwen3.7-max via Qwen Code /review_

File: packages/cli/src/commands/review/compose-review.ts [hunk 3]
```diff
@@ -590,16 +599,34 @@ export function composeReview(input: ComposeReviewInput): ComposeReviewResult {
...
+      const m = /^chunk (\d+)$/.exec(e);
```
Comment (wenshao) [inline packages/cli/src/commands/review/compose-review.ts:618]: **Suggestion** (non-blocking): The same `/^chunk (\d+)$/` parse-and-extract logic appears at three independent sites added by this diff (here, line 701, and line 816). If the subject format ever changes, all three must update in lockstep — miss one and that site silently stops recognising chunk subjects, routing them into the `named`/`callerNamed` bucket and rendering raw bookkeeping IDs in the posted body.

Consider extracting a small helper:
```ts
function parseChunkSubject(s: string): number | undefined {
  const m = /^chunk (\d+)$/.exec(s);
  return m ? Number(m[1]) : undefined;
}
```

File: packages/cli/src/commands/review/compose-review.ts [hunk 4]
```diff
@@ -799,6 +872,65 @@ export function composeReview(input: ComposeReviewInput): ComposeReviewResult {
...
+  let allKnown = planned.length > 0;
+  for (const id of uniq) {
+    const f = byId.get(id) ?? [];
+    if (f.length === 0) allKnown = false;
+    for (const p of f) {
+      if (!files.includes(p)) files.push(p);
+    }
+  }
+  if (allKnown && files.length <= 4) {
+    return {
```
Comment (wenshao) [inline packages/cli/src/commands/review/compose-review.ts:920]: **Suggestion** (non-blocking): `describeChunkGap([], planned)` with a non-empty plan returns `"the diff sections covering "` — trailing space, no file names. All three current callers guard with `.length > 0`, so this is unreachable today, but the function is exported and directly unit-tested.

A future caller that skips the guard posts a malformed sentence. Consider an early return (`if (uniq.length === 0) return { phrase: 'none of the diff', plural: false };`) or adding `files.length > 0` to this branch condition.

File: packages/cli/src/commands/review/compose-review.ts [hunk 5]
```diff
@@ -799,6 +872,65 @@ export function composeReview(input: ComposeReviewInput): ComposeReviewResult {
...
+      phrase: `the diff ${uniq.length === 1 ? 'section' : 'sections'} covering ${files.join(', ')}`,
+      plural: uniq.length > 1,
+    };
+  }
+  return {
+    phrase:
+      planned.length > 0
+        ? `${uniq.length} of the diff's ${planned.length} sections`
```
Comment (wenshao) [inline packages/cli/src/commands/review/compose-review.ts:928]: **Suggestion** (non-blocking): The count fallback uses `uniq.length` (all gap IDs, including any not present in `planned`) as the numerator against `planned.length`. If a caller-written bare `"chunk 99"` enters `bareIds` and 99 is not in the plan, the phrase reads "2 of the diff's 1 sections" — self-contradictory on the public PR page.

Reachability is narrow (caller entries typically carry file annotations, routing to `callerNamed`; coverage-derived entries always use planned IDs), but counting only IDs present in the plan when one is available would make the phrase robust:
```ts
const knownInGap = uniq.filter((id) => byId.has(id));
```

**Review by wenshao — COMMENTED** [review]
## Review — APPROVE (C=0)

Clean, well-scoped render-time fix. The two defects from the #7268 incident (contradictory opener, internal chunk IDs in the author-facing body) are both addressed correctly. All changes are additive and backward-compatible; the operator channel (stderr, remediation) is untouched.

Three suggestions below — none blocking.

**Verification:** Build passes across all 13 workspaces. No test failures attributable to this diff (pre-existing `shellExecutionService.ts` environment issue on macOS). Security, removed-behavior, and cross-file traces all clean.
